lib/playlist.js

```diff
@@ -0,0 +1,56 @@
+const GenreList = require('./genreList');
+const Song = require('./song');
+
+class Playlist{
+
+    // songs => list
+    constructor(songs){
+        this.songs = songs;
+        this.genres = {};
+        this.artists = new Set();
+
+        this.songs.forEach((song) => {
+            let genre = song.genre;
+            let feat = song.interpretes;
+            if (genre in this.genres){
+                this.genres[genre]++;
+            }
+            else{
+                this.genres[genre] = 1;
+            }
+
+            if (Array.isArray(feat)){
+                for (let a of feat){
+                    this.artists.add(a);
+                }
+            }
+            else{
+                this.debug = feat;
+                this.artists.add(feat);
+            }
+        })
+
+
+    };
+
+    getSortedGenres(){
+        let arr = Object.entries(this.genres).sort((a,b) => b[1] - a[1]);
```

Comment:
Haces esto *cada vez que se llama a la función*, mientras que this.genres es o inmutable o sólo puede cambiar de una forma determinada.

Reply:
Por otro lado, ¿seguro que esta es la mejor manera de hacer un sort de un hash en JavaScript?